src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -0,0 +1,316 @@
+            self.env.notification_manager(),
+            &self.metrics,
+        );
+        let mut new_version_delta = version.new_delta();
+
+        let target_compaction_group_id = {
+            let mut config = self
+                .compaction_group_manager
+                .read()
+                .await
+                .try_get_compaction_group_config(group_1)
+                .unwrap()
+                .compaction_config()
+                .deref()
+                .clone();
+
+            {
+                // update config
+                config.split_weight_by_vnode = 0;
+            }
+
+            // merge right_group_id to left_group_id and remove right_group_id
+            new_version_delta.group_deltas.insert(
+                left_group_id,
+                GroupDeltas {
+                    group_deltas: vec![GroupDelta::GroupMerge(PbGroupMerge {
+                        left_group_id,
+                        right_group_id,
+                    })],
+                },
+            );
+            left_group_id
+        };
+
+        // TODO: remove compaciton group_id from state_table_info
+        // rewrite compaction_group_id for all tables
+        new_version_delta.with_latest_version(|version, new_version_delta| {
+            for table_id in combine_member_table_ids {
+                let table_id = TableId::new(table_id.table_id());
+                let info = version
+                    .state_table_info
+                    .info()
+                    .get(&table_id)
+                    .expect("have check exist previously");
+                assert!(new_version_delta
+                    .state_table_info_delta
+                    .insert(
+                        table_id,
+                        PbStateTableInfoDelta {
+                            committed_epoch: info.committed_epoch,
+                            safe_epoch: info.safe_epoch,
+                            compaction_group_id: target_compaction_group_id,
+                        }
+                    )
+                    .is_none());
+            }
+        });
+
+        {
+            let mut compaction_group_manager = self.compaction_group_manager.write().await;
+            let mut compaction_groups_txn = compaction_group_manager.start_compaction_groups_txn();
+
+            // for metrics reclaim
+            {
+                let right_group_max_level = new_version_delta
+                    .latest_version()
+                    .get_compaction_group_levels(right_group_id)
+                    .levels
+                    .len();
+
+                remove_compaction_group_in_sst_stat(
+                    &self.metrics,
+                    right_group_id,
+                    right_group_max_level,
+                );
+            }
+
+            new_version_delta.pre_apply();
+
+            // purge right_group_id
+            compaction_groups_txn.purge(HashSet::from_iter(get_compaction_group_ids(
```

Comment:
Can we directly call `compaction_groups_txn.remove(...)`?

Reply:
Good Catch, We've already triggered Group Destory in the Merge Delta process, so we really don't need to purge again here.

Reply:
Can you explain more on this? I think here we are manipulating the `CompactionGroup` metadata in `CompactionGroupManager`, not in `HummockVersion`. Why is this related to GroupDestroy delta?